python/GafferSceneUI/RenderPassEditor.py

```diff
@@ -612,6 +612,50 @@ def __disableEdits( self, pathListing ) :
 					if isinstance( source, ( Gaffer.TweakPlug, Gaffer.NameValuePlug ) ) :
 						source["enabled"].setValue( False )
 
+	def __selectedSetExpressions( self, pathListing ) :
+
+		result = set()
+
+		with Gaffer.Context( self.getContext() ) as context :
+			renderPassPath = pathListing.getPath().copy()
+			for columnSelection, column in zip( pathListing.getSelection(), pathListing.getColumns() ) :
+				if (
+					not columnSelection.isEmpty() and (
+						not isinstance( column, _GafferSceneUI._RenderPassEditor.OptionInspectorColumn ) or
+						not (
+							Gaffer.Metadata.value( "option:" + column.inspector().name(), "ui:scene:acceptsSetName" ) or
+							Gaffer.Metadata.value( "option:" + column.inspector().name(), "ui:scene:acceptsSetNames" ) or
+							Gaffer.Metadata.value( "option:" + column.inspector().name(), "ui:scene:acceptsSetExpression" )
+						)
+					)
+				) :
+					# We only return set expressions if all selected paths are in
+					# columns that accept set names or set expressions.
+					return []
+
+				for path in columnSelection.paths() :
+					renderPassPath.setFromString( path )
+					renderPassName = renderPassPath.property( "renderPassPath:name" )
+					if not renderPassName :
+						# We only return set expressions if all selected paths are render passes.
+						return []
+
+					context["renderPass"] = renderPassName
+					inspection = column.inspector().inspect()
+					if inspection.value() :
+						result.add( inspection.value().value )
```

Comment:
What do you think to using `column.cellData( renderPassPath ).value` here? It would do all the inspection and not-a-render-pass checks for us, and would mean we wouldn't need to do any context management of our own (we could remove L619).

There might also be a future where PathListingWidget could give us the current CellData for a cell directly, so we weren't doing any computations on the UI thread. So we'd always be doing the operation with the value you see and not some value that might be pending a background update.

Reply:
Good idea! `column.cellData( renderPassPath ).value` does simplify things and I've switched to using it in 49591c6.